src/astro/databases/google/bigquery.py

```diff
@@ -236,3 +257,177 @@ def gs_to_bigquery(
+
+        self.project_id = project_id
+        self.poll_duration = poll_duration
+        self.native_support_kwargs = native_support_kwargs
+        self.kwargs = kwargs
+
+    def run(self):
+        """
+        Algo to run S3 to Bigquery datatransfer
+        """
+        transfer_config_id = self.create_transfer_config()
+        try:
+            # Manually run a transfer job using previously created transfer config
+            run_id = self.run_transfer_now(transfer_config_id)
+
+            # Poll Bigquery for status of transfer job
+            run_info = self.get_transfer_info(
+                run_id=run_id, transfer_config_id=transfer_config_id
+            )
+
+            # Note - Super set of states that indicate the job is running.
+            # This needs to be a super set as this if we miss on any running state, code will go into infinite loop.
+            running_states = [TransferState.PENDING, TransferState.RUNNING]
+
+            while run_info.state in running_states:
+                run_info = self.get_transfer_info(
+                    run_id=run_id, transfer_config_id=transfer_config_id
+                )
+                time.sleep(self.poll_duration)
+
+            if run_info.state != TransferState.SUCCEEDED:
+                raise ValueError(run_info.error_status)
+        finally:
+            # delete transfer config created.
+            self.delete_transfer_config(transfer_config_id)
+
+    @staticmethod
+    def get_transfer_config_id(config: TransferConfig) -> str:
+        """
+        Extract transfer_config_id from TransferConfig object
+        """
+        # ToDo: Look for a native way to extract 'transfer_config_id'
+        # name - "projects/103191871648/locations/us/transferConfigs/6302bf19-0000-26cf-a568-94eb2c0a61ee'.
+        # We need extract transferConfigs which is at the end of string.
+        tokens = config.name.split("transferConfigs/")
+        return str(tokens[-1])
+
+    @staticmethod
+    def get_run_id(config: TransferRun) -> str:
+        """
+        Extract run_id from TransferRun object
+        """
+        # ToDo: Look for a native way to extract 'run_id'
+        # config.runs[0].name - "projects/103191871648/locations/us/
+        # transferConfigs/62d38894-0000-239c-a4d8-089e08325b54/runs/62d6a4df-0000-2fad-8752-d4f547e68ef4'.
+        # We need extract transferConfigs which is at the end of string.
+        tokens = config.runs[0].name.split("runs/")
+        return str(tokens[-1])
+
+    def create_transfer_config(self):
+        """Create bigquery transfer config on cloud"""
+        s3_params = {
+            "destination_table_name_template": self.target_table.name,
+            "data_path": self.source_file.path,
+            "access_key_id": self.s3_login,
+            "secret_access_key": self.s3_password,
+            "file_format": self.s3_file_type,
+        }
+        s3_params.update(self.native_support_kwargs)
+
+        s3_params.update(self.kwargs)
```

Comment:
Why do we have both `kwargs` and `native_support_kwargs`?

Reply:
was a mistake will remove it.